### PR TITLE
Add clj kondo

### PR DESCRIPTION
**Pull request by pfeodrippe**
Fixes https://github.com/overtone/overtone/issues/485.

Remove all linting errors \o/

Also, some bugs were fixed.

**Before**
Screen Shot 2022-09-28 at 12 02 14 AM

**Now**
Screen Shot 2022-09-27 at 11 55 08 PM

### CODE REVIEW

File: src/overtone/helpers/synth.clj [hunk 1]
```diff
@@ -10,9 +10,14 @@
   (when-not (valid-synth-node-pos pos)
     (throw (IllegalArgumentException. (str "Invalid synth node target. Was expecting one of " valid-synth-node-pos ", found: " pos)))))
 
+(def ^:dynamic *target-pos* nil)
```
Comment (hlolli) [inline src/overtone/helpers/synth.clj:13]: this is global scoped, could it be better to create an atom inside extract-target-pos-args* body?

File: src/overtone/inst/sampler.clj [hunk 2]
```diff
@@ -28,14 +28,16 @@
...
         env (- (env-gen:ar (perc 0.01 dur) 1 1 0 1 FREE) 0.001)]
-    (out out (* (pan2 (* grain env) pan) amp))))
+    (out out-bus (* (pan2 (* grain env) pan) amp))))
 
-(defn test-flute []
-  (defonce flute (load-sample (load-sample "/home/rosejn/projects/overtone/instruments/samples/flutes/flutter-flute-1.wav")))
-  (hit flute :dur 2.0))
+;; TODO: hit is not defined
+#_(defn test-flute []
+    (defonce flute (load-sample (load-sample "/home/rosejn/projects/overtone/instruments/samples/flutes/flutter-flute-1.wav")))
```
Comment (hlolli) [inline src/overtone/inst/sampler.clj:40]: we could replace this with freesound url eventually

**Review by hlolli — APPROVED** [review]
very good, since you're in the mood of refactoring, I'd love to see the use of :use disappear eventually. Good job!

**Comment by hlolli** [comment]
Btw if you can add to CHANGELOGS, it would be nice!

File: src/overtone/helpers/synth.clj (hunk 1 above)
Comment (pfeodrippe) [reply src/overtone/helpers/synth.clj:13]: Oops, I will remove this tonight, some leftovers from an experiment I was doing, thanks o/

**Comment by pfeodrippe** [comment]
> very good, since you're in the mood of refactoring, I'd love to see the use of :use disappear eventually. Good job!

Yeah, hopefully in the future 🙏 Clj kondo is the first step in making it more static-friendly, I’ve learnt a lot about the project namespaces, there are a lot of good ideas there!

**Comment by pfeodrippe** [comment]
> Btw if you can add to CHANGELOGS, it would be nice!

Will do, thanks, and before doing any proper release, will do a small scan in the previous commits to add the most important to the change log.

File: src/overtone/inst/sampler.clj (hunk 2 above)
Comment (pfeodrippe) [reply src/overtone/inst/sampler.clj:40]: Yes o/

**Comment by pfeodrippe** [comment]
Thanks for the time for reviewing it, Hlöðver :D, I will remove the dynamic vars and merge it tonight.

**Pull request merged**